lib/data-serializer.js

```diff
@@ -8,12 +8,23 @@ class DataSerializer {
   /**
    * Get latest split and segment data and return a script containing serialized strings.
    */
-  generateSerializedDataScript () {
-    const cachedSerializedData = this.poller.cache.serializedData ? this.poller.cache.serializedData : '{}'
+  generateSerializedDataScript (splits = []) {
+    const serializedData = splits.length > 0
+      ? this._getSomeSerializedData(splits)
+      : this._getAllSerializedData()
+
     return `<script>
-      window.__splitCachePreload = ${cachedSerializedData};
+      window.__splitCachePreload = ${serializedData};
     </script>`
   }
+
+  _getSomeSerializedData (splits) {
```

Comment:
```suggestion
  _getFilteredSerializedData (splits) {
```
Should we make the method name clearer?